src/sonic-yang-models/yang-models/sonic-fine-grained-ecmp.yang

```diff
@@ -0,0 +1,136 @@
+    }   
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC Fine Grained ECMP";
+
+    revision 2023-01-19 {
+        description "Initial revision.";
+    }
+
+
+    container sonic-fine-grained-ecmp {
+
+        container FG_NHG { 
+
+            description "FG_NHG part of config_db.json";
+
+            list FG_NHG_LIST{
+                key "name";
+
+                leaf name{
+                    type string;
+                    description "FG_NHG group name";
+                }
+
+                leaf bucket_size{
+                    type uint16;
+                    mandatory true;
+                    description "total hash bucket size desired, 
+                    recommended value of Lowest Common Multiple 
+                    of 1..{max # of next-hops}";
+                }
+
+                leaf match_mode{
+                    type string{
+                        pattern "(route-based|nexthop-based)"{
+                            error-message "Invalid FG_NHG match mode";
+                        }
+                    }
+                    mandatory true;
+                    description " The filtering method used to identify
+                    when to use Fine Grained vs regular route handling.
+                    nexthop-based looks to next-hop IP to filter routes
+                    and uses fine grained ecmp when nexthop IPs matches
+                    FG_NHG_MEMBER IPs. route-based looks to prefix to
+                    filter routes, and uses fine grained ecmp when the
+                    route prefix matches the FG_NHG_PREFIX prefix.";
+                }
+            }
+             /* end of list FG_NHG_LIST */
+        }
+        /* end of container FG_NHG */
+
+        container FG_NHG_PREFIX{
+
+            description "FG_NHG_PREFIX part of config_db.json";
+
+            list FG_NHG_PREFIX_LIST{
+
+                key "ip_prefix";
+
+                leaf ip_prefix{
+                    type stypes:sonic-ip-prefix;
+                    description "FG_NHG_PREFIX for which FG behavior is
+                    desired";
+                }
+
+                leaf FG_NHG{
+                    type string;
+                    mandatory true;
+                    description "Fine Grained next-hop group name";
+                }
+            }
+            /* end of list FG_NHG_PREFIX_LIST */
+        }
+        /* end of container FG_NHG_PREFIX */
+
+        container FG_NHG_MEMBER{
+
+            description "FG_NHG_MEMBER part of config_db.json";
+
+            list FG_NHG_MEMBER_LIST{
+
+                key "next_hop_ip";
+
+                leaf next_hop_ip{
+                    type inet:ip-address;
+                    description "FG_NHG next-hop-ip member associated
+                    with prefix";
+                }
+
+                leaf FG_NHG{
+                    type string;
```

Comment:
Needs to be a leafref to FG_NHG list above

Reply:
changed to leafref